src/Embed.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+
+/**
+ * Recurse the clicked target to see if we can work out whether an anchor (or button) was clicked or not. Return the
+ * anchor clicked if one is found, otherwise null.
+ */
+// @ts-ignore-next-line
+const findElement = (element: any): Element | null => {
+  // Is there a current element (or a provided parent)?
+  if (!element) {
+    return null;
+  }
+  // Is the current element something clickable?
+  if (['a', 'button'].includes(element.tagName.toLowerCase())) {
+    return element;
+  }
+  // Check if parent element is...
+  return findElement(element.parentElement);
+};
+
+function Embed() {
+  const [url, setUrl] = useState('');
+  const [visible, setVisible] = useState(false);
+
+  // Register callback on load to get CSS involved.
+  useEffect((): void => {
+    const stylesheetUrl = process.env.NODE_ENV === 'development'
+      ? '/static/css/embed.css'
+      : 'https://cdn.chec.io/v3/embed.css';
+
+    const stylesheet = document.createElement('link');
+    stylesheet.setAttribute('rel', 'stylesheet');
+    stylesheet.setAttribute('type', 'text/css');
+    stylesheet.setAttribute('href', stylesheetUrl);
+
+    document.getElementsByTagName('head')[0].appendChild(stylesheet);
+  }, []);
+
+  // Register click handler that will look for any anchors in the page clicked that contain
+  // a "data-chec-embed-url" attribute, and if that is found then use it to open an embed modal
+  useEffect(() => {
+    const clickSpy = (e: Event) => {
+      const anchor = findElement(e.target);
+
+      // It wasn't an anchor that was clicked, do nothing.
+      if (!anchor) {
+        return;
+      }
+
+      // It was an anchor, check if it has the required data attribute
+      const embedUrl = anchor.getAttribute('data-chec-embed-url');
+      if (!embedUrl) {
+        return;
+      }
+
+      // It has the required attribute, use it!
+      setUrl(embedUrl);
+      setVisible(true);
+
+      // Prevent whatever the click would've done
+      e.preventDefault();
+      return false;
+    };
+
+    window.addEventListener('click', clickSpy);
+    return () => {
+      window.removeEventListener('click', clickSpy);
+    };
+  }, []);
+
+  // If the modal is open, then bind the escape key to close it. Remove the handler when closing it.
+  useEffect(() => {
+    if (!visible) {
+      return;
+    }
+
+    const closeSpy = (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
+        setVisible(false);
+      }
+    };
+
+    window.addEventListener('keyup', closeSpy);
+    return () => {
+      window.removeEventListener('keyup', closeSpy);
+    };
+  }, [visible]);
+
+  if (!visible || !url) {
+    return null;
+  }
+
+  return (
+    <div className="chec-embed">
+      <div className="chec-embed__container">
+        <iframe src={url} width="100%" height="100%" title="Buy now" className="chec-embed__frame">
+          Your browser does not support frames, please visit {url} directly.
```

Comment:
Do you want to make `{url}` actually a link here?

Reply:
I think it's very unlikely that browsers won't support iframes, so no